# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmIonCompile.cpp
# Commit: 0e268b8d01fe
# Full Hash: 0e268b8d01fe869fc9c488e4dc627b5962a9ea43
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-02 04:10:47
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.0: Force Ion to preserve its tls. r=lth
#   
#   Let's see following code and let's assume that wasm-compiler is ion:
#   
#   call foo
# ==============================================================================

diff -r 062970b20339 -r 0e268b8d01fe js/src/wasm/WasmIonCompile.cpp
--- a/js/src/wasm/WasmIonCompile.cpp	Thu Oct 01 11:35:23 2020 +0000
+++ b/js/src/wasm/WasmIonCompile.cpp	Thu Oct 01 04:54:42 2020 +0000
@@ -1571,7 +1571,7 @@
     }
 
     if (values.empty()) {
-      curBlock_->end(MWasmReturnVoid::New(alloc()));
+      curBlock_->end(MWasmReturnVoid::New(alloc(), tlsPointer_));
     } else {
       ResultType resultType = ResultType::Vector(funcType().results());
       ABIResultIter iter(resultType);
@@ -1603,7 +1603,7 @@
         } else {
           MOZ_ASSERT(iter.remaining() == 1);
           MOZ_ASSERT(i + 1 == values.length());
-          curBlock_->end(MWasmReturn::New(alloc(), values[i]));
+          curBlock_->end(MWasmReturn::New(alloc(), values[i], tlsPointer_));
         }
       }
     }
